ehcache/src/main/scala/scalacache/ehcache/EhcacheCache.scala

```diff
@@ -28,12 +28,17 @@ class EhcacheCache(underlying: Ehcache)
    * @return the value, if there is one
    */
   override def get[V](key: String)(implicit codec: Codec[V, InMemoryRepr]) = {
-    val result = for {
-      e <- Option(underlying.get(key))
-      v <- Option(e.getObjectValue.asInstanceOf[V])
-    } yield v
-    logCacheHitOrMiss(key, result)
-    Future.successful(result)
+
+    val valueObj = underlying.get(key)
+
+    if (valueObj != null) {
```

Comment:
EhCache is a bit tricky. There are multiple cases to handle here:

* `underlying.get(key)` returns `null`
* It returns a non-null element, but `elem.getObjectValue` is `null`

I'd suggest this:

```scala
val result = {
  val elem = underlying.get(key)
  if (elem == null) None
  else Option(elem.getObjectValue.asInstanceOf[V])
}
logCacheHitOrMiss(key, result)
Future.successful(result)
```